indra/llxml/llxmlnode.cpp

```diff
@@ -653,31 +653,67 @@ bool LLXMLNode::updateNode(
 // static
 bool LLXMLNode::parseFile(const std::string& filename, LLXMLNodePtr& node, LLXMLNode* defaults_tree)
 {
-	// Read file
 	LL_DEBUGS("XMLNode") << "parsing XML file: " << filename << LL_ENDL;
+
+ #ifdef _WIN64
+    // Try to read from resource
+    std::string app_dir = gDirUtilp->getAppRODataDir() + '\\';
+    // E.g., if finename starts with "C:\\Program Files\\SecondLifeViewer\\"
+    if (filename.substr(0, app_dir.size()) == app_dir)
+    {
+        // Convert the rest of filename to resource_id
+        // E.g., "app_settings\\settings_files.xml" => L"IDR_APP_SETTINGS_SETTINGS_FILES_XML"
+        std::string filepath = filename.substr(app_dir.size());
+        LLStringUtil::replaceChar(filepath, '\\', '_');
+        LLStringUtil::replaceChar(filepath, '%', '_');
+        LLStringUtil::replaceChar(filepath, '.', '_');
+        LLStringUtil::toUpper(filepath);
+        filepath = "IDR_" + filepath;
+        std::wstring resource_id(filepath.begin(), filepath.end());
+
+        HMODULE hmodule = GetModuleHandleW(nullptr);
+        HRSRC hresource = FindResourceW(hmodule, resource_id.c_str(), L"XML");
+        if (hresource)
+        {
+            HGLOBAL hglobal = LoadResource(hmodule, hresource);
+            if (hglobal)
+            {
+                const char* text = (const char*)LockResource(hglobal);
+                if (text)
+                {
+                    size_t size = SizeofResource(nullptr, hresource);
+                    if (LLXMLNode::parseBuffer(text, size, node, defaults_tree))
+                        return true;
+                }
+            }
+        }
+    }
+#endif
+
+	// Try to read from file
```

Comment:
This doesn't cover all files, for example removing commands.xml still causes a crash.

I'm not against the approach of moving files into resources, but since it can't cover some files at all (like updater and plugins) I would rather have viewer complain to user about a bad install than trying to fix consequences one file at a time.

Reply:
There is still fallback code below, so it will likely be fine.

Reply:
> This doesn't cover all files
I do not pretend on this, just want to make the viewer more stable and fast

Reply:
>There is still fallback code below, so it will likely be fine.
I mean that:
1. Instead of failing on a first file with LL_ERRS and a visual warning in D591 it will just fail at next one and not nessesaryly with an LL_ERRS and defientely without a warning. If you are doing this, than move the visual warning to next file that is most likely to crash viewer. 
2. Files were moved into resources yet remain unused. In case of settings.xml it means extra 500kb of weight.

Reply:
>There is still fallback code below, so it will likely be fine.
Actually this is going to break preview tools... It looks like it copies all files (including floaters) which means viewer will load resources first and live-time modification via previe tool wouldn't work.

Reply:
Oh. That's a problem.

Reply:
>There is still fallback code below, so it will likely be fine.

Actually this is going to break preview tools... It looks like it copies all files (including floaters) which means viewer will load resources first and live-time modification via previe tool wouldn't work.